File: workspace/solutions/go/_0/_0/_0/_1/s_0001_01/s_0001_01.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

package _0001_01

// @lc code=start
func twoSum(nums []int, target int) []int {
	_map := map[int]int{}
	for i := 0; ; i++ {
		num := nums[i]
		diff := target - num
		if j, ok := _map[diff]; ok {
			return []int{j, i}
		}
		_map[num] = i
	}
}

// @lc code=end

var TwoSum = twoSum

File: workspace/solutions/kotlin/lib/src/main/kotlin/_0/_0/_0/_1/s_0001_01/s_0001_01.kt
/*
 * @lc app=leetcode.cn id=1 lang=kotlin
 *
 * [1] 两数之和
 */

package _0._0._0._1.s_0001_01

// @lc code=start
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()
        nums.forEachIndexed { i, num ->
            val diff = target - num
            val j = map.get(diff)
            if (j != null) {
                return intArrayOf(j, i)
            }
            map[num] = i
        }
        return intArrayOf()
    }
}
// @lc code=end

public class s_0001_01 : Solution
